### ISSUE TITLE
TestCheckSuite/TestCheckAbort failure

### ISSUE THREAD

**Issue by bradbeam** [talos-systems/talos#712]
```
=== RUN TestCheckSuite    229s
334    === RUN TestCheckSuite/TestCheckAbort    229s
335    === RUN TestCheckSuite/TestHealthChange    229s
336    === RUN TestCheckSuite/TestHealthy    229s
337    === RUN TestCheckSuite/TestInitialState    229s
338    --- FAIL: TestCheckSuite (0.30s)    229s
339    --- FAIL: TestCheckSuite/TestCheckAbort (0.05s)    229s
340    require.go:347:    229s
341    Error Trace:    health_test.go:146    229s
342    Error: Should be false    229s
343    Test: TestCheckSuite/TestCheckAbort    229s
344    --- PASS: TestCheckSuite/TestHealthChange (0.10s)    229s
345    --- PASS: TestCheckSuite/TestHealthy (0.10s)    229s
346    --- PASS: TestCheckSuite/TestInitialState (0.05s)    229s
347    FAIL    229s
348    coverage: 98.2% of statements    229s
349    FAIL    github.com/talos-systems/talos/internal/app/init/pkg/system/health    0.311s
```

**Comment by smira** [comment]
I believe this also happens under high load, like with many tests trying to address concurrent code, it relies on `time.Sleep` for other goroutines to advance in execution. I will take a look to see if there's a way to improve it

**Comment by smira** [comment]
Hopefully fixed via #713 (but not 100% sure)
